src/reducers/importText.ts

```diff
@@ -157,22 +160,55 @@ const importText = (state: State, { path, text, lastUpdated, preventSetCursor, r
 
   }
   else {
+
     const json = isRoam ? roamJsonToBlocks(JSON.parse(convertedText)) : convertHTMLtoJSON(convertedText)
-    const imported = importJSON(state, simplePath, json, { lastUpdated, skipRoot })
 
-    return reducerFlow([
+    const uuid = createId()
 
-      updateThoughts(imported),
+    // Note: Create a dummy thought and then import new thoughts into its context and then collpase it . Since collapse uses existingThoughtMove it merges imported thoughts.
+    const updatedState = newThought(state, {
+      at: simplePath,
+      insertNewSubthought: true,
+      value: uuid
+    })
 
-      // restore the selection to the last imported thought on the first level
-      !preventSetCursor
-        ? setCursor({
-          path: imported.lastImported,
-          offset: headValue(imported.lastImported).length
-        })
-        : null,
+    const destEmpty = destThought.value === '' && getAllChildren(state, pathToContext(simplePath)).length === 0
 
-    ])(state)
+    // New dummy thoughts for collapsing
+    const newDummyThought = getAllChildren(updatedState, pathToContext(simplePath)).find(child => child.value === uuid)
```

Comment:
It's not super important, but you should probably store `pathToContext(simplePath)` in a variable since you call it multiple times.